### PR TITLE
Cm used orders

### PR DESCRIPTION
**Pull request by chrismellor08**
What Christopher Patrick mellor, HE AND HIMSELF ONLY changed:
Created a function and helper function that will only allow a user to delete a product if it does not already exist on an existing order

To test,
open your DB browser and create some rows in the join table for experimental deletion. Enter a few products for the same order_id. Make sure that there are products that do and do not exist on your created productOrders.
Use postman to delete them based on their product_id
Also- try to delete a product that does exist on a productOrder and make sure that it does not delete

### CODE REVIEW

File: controllers/productsCtrl.js [hunk 1]
```diff
@@ -45,4 +45,26 @@ module.exports.putProduct = (req, res, next) => {
     .catch((err)=>{
         next(err);
     })
-}
\ No newline at end of file
+}
+
+module.exports.deleteAProduct = ({params: {id}}, res, next) => {
+    console.log(id);
```
Comment (mozecode) [inline controllers/productsCtrl.js]: Please remove console.logs. :)

File: models/Product.js [hunk 2]
```diff
@@ -53,6 +53,23 @@ module.exports ={
...
+        return new Promise((resolve, reject)=>{
+            //if product_type_id exists inside products, don't delete the product type
+            db.all(`SELECT * FROM products as p
+            WHERE NOT EXISTS
+            (
+               SELECT * FROM productOrders as po
+               WHERE p.product_id = po.product_id
+            )`, (err, data)=>{
+                if (err) return reject(err);
+                console.log(data);
```
Comment (mozecode) [inline models/Product.js]: Please remove console.logs. :)

**Comment by mozecode** [comment]
👍 You rock!  Works as expected.

**Comment by Rian501** [comment]
💩 yes

**Pull request merged**